ontology.ttl

```diff
@@ -160,7 +160,28 @@ hobbit:organizer
 		rdfs:label "Organizer"^^xsd:string ;
 		rdfs:domain hobbit:Challenge ;
 		rdfs:range xsd:string .
-				
+
+hobbit:registrationCutoffDate
+		rdf:type owl:DatatypeProperty ;
+		rdfs:comment "Date after which no more changes to the systems participating can be made. No more experiments will run after this date."^^xsd:string ;
+		rdfs:label "Registration Cutoff Date"^^xsd:string ;
+		rdfs:domain hobbit:Challenge ;
+		rdfs:range xsd:dateTime .
+
+hobbit:executionPeriod
+		rdf:type owl:DatatypeProperty ;
+		rdfs:comment "Time between consecutive executions."^^xsd:string ;
+		rdfs:label "Period"^^xsd:string ;
+		rdfs:domain hobbit:Challenge ;
+		rdfs:range xsd:duration .
```

Comment:
I don't know enough about this data type to examine whether it is a good choice...
Can you please check whether there is a good representation for it in Java? Otherwise we might use an easier datatype, e.g., `xsd:int` and the meaning "time in hours".

Reply:
Looks like there is [org.apache.jena.datatypes.xsd.XSDDuration](https://jena.apache.org/documentation/javadoc/jena/org/apache/jena/datatypes/xsd/XSDDuration.html) which can be converted (using `toString` representation) to [java.time.Duration](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html).

But `java.util.Calendar` we use doesn't seem to work as is with such `Duration` objects without additional conversions. It has an `add` method where we can specify a time unit and the amount, so yes, using integer value (and maybe optional time unit) may be easier to implement (but less universal).

Reply:
The core handling this will look like this:

```java
Duration executionPeriod = RdfHelper.getDurationValue(challengesModel, challenge, HOBBIT.executionPeriod);
dateOfNextExecution.add(Calendar.MILLISECOND, executionPeriod.toMillis());
```

```java
    public static Duration getDurationValue(Model model, Resource subject, Property predicate) {
        if (model == null) {
            return null;
        }
        Literal literal = getLiteral(model, subject, predicate);
        if (literal != null) {
            try {
                Object o = XSDDatatype.XSDduration.parse(literal.getString());
                if (o instanceof XSDDuration) {
                    return Duration.parse(o.toString());
                }
            } catch (Exception e) {
                // nothing to do
                LOGGER.debug("Couldn't parse " + XSDDatatype.XSDduration.getURI() + ". Returning null.", e);
            }
        }
        return null;
    }
```

Need to test it when everything will be in place, so we can merge this later.

Reply:
Duration can be represented with `Java.Time.Duration`. While not ideal when used with `Calendar`, looks like it works fine (code already exists in `feature/repeatable-challenge` branches).